lib/pbench/cli/agent/commands/results/__init__.py

```diff
@@ -1,94 +1,213 @@
+"pbench-move-results"
+
 import os
-import requests
+import click
+
+# import requests
 import shutil
-import sys
+
+# import sys
 import tempfile
 
 from pbench.agent import PbenchAgentConfig
+from pbench.agent.base import BaseCommand
 from pbench.agent.results import MakeResultTb, CopyResultTb
+from pbench.common.exceptions import (
+    BadConfig,
+    BadMDLogFormat,
+    FileUploadError,
+)
 from pbench.common.logger import get_pbench_logger
+from pbench.cli.agent import pass_cli_context
+from pbench.cli.agent.options import common_options
 
 
-def move_results(ctx, _user, _prefix, _show_server):
-    config = PbenchAgentConfig(ctx["args"]["config"])
-    logger = get_pbench_logger("pbench", config)
+class MoveResults(BaseCommand):
+    def __init__(self, context):
+        super().__init__(context)
 
-    controller = os.environ.get("_pbench_full_hostname")
-    if not controller:
-        logger.error("Missing controller name (should be 'hostname -f' value)")
-        sys.exit(1)
+    def execute(self):
+        try:
+            config = PbenchAgentConfig(self.context.config)
+        except BadConfig:
+            raise
 
-    results_webserver = config.results.get("webserver")
-    if not results_webserver:
-        logger.error(
-            "No web server host configured from which we can fetch the FQDN of the host to which we copy/move results"
-        )
-        logger.debug("'webserver' variable in 'results' section not set")
+        logger = get_pbench_logger("pbench-agent", config)
 
-    server_rest_url = config.results.get("server_rest_url")
-    response = requests.get(f"{server_rest_url}/host_info")
-    if response.status_code not in [200, 201]:
-        logger.error(
-            "Unable to determine results host info from %s/host_info", server_rest_url
-        )
-        sys.exit(1)
-    if response.text.startswith("MESSAGE"):
-        message = response.text.split("===")[1]
-        logger.info("*** Message from sysadmins of %s:", results_webserver)
-        logger.info("***\n*** %s", message)
-        logger.info("***\n*** No local actions taken.")
-        sys.exit(1)
-    results_path_prefix = response.text.split(":")[1]
-    if not results_path_prefix:
-        logger.error(
-            "fetch results host info did not contain a path prefix: %s", response.text
-        )
-        sys.exit(1)
+        if self.context.user:
+            logger.warning("Option --user is no longer being used")
+        if self.context.prefix:
+            logger.warning("Option --prefix is no longer being used")
```

Comment:
I suspect we should be writing this output as a message to `stderr` instead of just to the log, which most users probably won't see...

Reply:
yeah that would be good :)